packages/install/first_block.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 The go-daylight Authors
+// This file is part of the go-daylight library.
+//
+// The go-daylight library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-daylight library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-daylight library. If not, see <http://www.gnu.org/licenses/>.
+
+package install
+
+import (
+	"encoding/hex"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/AplaProject/go-apla/packages/conf"
+	"github.com/AplaProject/go-apla/packages/consts"
+	"github.com/AplaProject/go-apla/packages/converter"
+	"github.com/AplaProject/go-apla/packages/crypto"
+	"github.com/AplaProject/go-apla/packages/parser"
+	"github.com/AplaProject/go-apla/packages/utils"
+)
+
+const fileMode = 0644
+
+// ErrFirstBlockHostIsEmpty host for first block is not specified
+var ErrFirstBlockHostIsEmpty = errors.New("FirstBlockHost is empty")
+
+func createKeyPair(privFilename, pubFilename string) (priv, pub string, err error) {
+	priv, pub, err = crypto.GenHexKeys()
+	if err != nil {
+		return
+	}
+
+	err = createFile(privFilename, []byte(priv))
+	if err != nil {
+		return
+	}
+
+	err = createFile(pubFilename, []byte(pub))
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func createFile(filename string, data []byte) error {
+	err := ioutil.WriteFile(filename, data, fileMode)
+	if err != nil {
+		log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("writing file")
+		return err
+	}
+	return nil
+}
+
+func getPublicKeyAndCreateKeyPair(value *string, privFilename, pubFilename string) ([]byte, error) {
```

Comment:
There should be no functions with "AND" in the name, because single function - single responsibility